### PR TITLE
Apply container patches based on package NVR (SOFTWARE-4629)

### PR DESCRIPTION
**Pull request by RyanEweSeng**

### CODE REVIEW

**Comment by brianhlin** [comment]
@RyanEweSeng I didn't look over the scripts themselves but let's throw them into a `bin/` directory in this repository then `COPY` the entirety of `bin/` into `/usr/local/bin` inside the container

File: pkg-cmp-between.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+USAGE="Usage: pkg-cmp-between.sh <RPM> <EVR1> <EVR2>"
+
+RPM=$1
+EVR1=$2
+EVR2=$3
+
+if [[ $# -ne 3 ]]; then
+    echo "Expected 3 args, got $#"
+    echo "$USAGE"
+    exit
```
Comment (matyasselmeci) [inline pkg-cmp-between.sh:15]:
```suggestion
    exit 2
```
The standard exit code for incorrect usage is 2.

File: pkg-cmp-gt.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+USAGE="Usage: pkg-cmp-gt.sh <RPM> <EVR>"
+
+RPM=$1
+EVR=$2
+
+if [[ $# -ne 2 ]]; then
+    echo "Expected 2 args, got $#"
+    echo "$USAGE"
+    exit
```
Comment (matyasselmeci) [inline pkg-cmp-gt.sh:14]:
```suggestion
    exit 2
```

File: pkg-cmp-gt.sh [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+fi
+
+RPM_EVR=$(rpm -q --queryformat '%{EPOCH}:%{VERSION}-%{RELEASE}\n' $RPM) # get EVR of the RPM
+RPM_EVR=${RPM_VER/(none)/0} # no epoch will print (none) for the epoch field so we change it to 0
```
Comment (matyasselmeci) [inline pkg-cmp-gt.sh:18]:
```suggestion
RPM_EVR=${RPM_EVR/(none)/0} # no epoch will print (none) for the epoch field so we change it to 0
```

File: pkg-cmp-between.sh [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+fi
+
+RPM_EVR=$(rpm -q --queryformat '%{EPOCH}:%{VERSION}-%{RELEASE}\n' $RPM) # get EVR of the RPM
+RPM_EVR=${RPM_VER/(none)/0} # no epoch will print (none) for the epoch field so we change it to 0
```
Comment (matyasselmeci) [inline pkg-cmp-between.sh:19]:
```suggestion
RPM_EVR=${RPM_EVR/(none)/0} # no epoch will print (none) for the epoch field so we change it to 0
```

File: pkg-cmp-between.sh [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+rpmdev-vercmp RPM_EVR EVR1 # check if RPM is newer/equal to EVR1 and store the return val
```
Comment (matyasselmeci) [inline pkg-cmp-between.sh:21]:
```suggestion
rpmdev-vercmp "$RPM_EVR" "$EVR1" # check if RPM is newer/equal to EVR1 and store the return val
```

File: pkg-cmp-between.sh [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+EVR1=$2
+EVR2=$3
+
+if [[ $# -ne 3 ]]; then
+    echo "Expected 3 args, got $#"
+    echo "$USAGE"
+    exit
+fi
+
+RPM_EVR=$(rpm -q --queryformat '%{EPOCH}:%{VERSION}-%{RELEASE}\n' $RPM) # get EVR of the RPM
```
Comment (matyasselmeci) [inline pkg-cmp-between.sh:18]:
```suggestion
RPM_EVR=$(rpm -q --queryformat '%{EPOCH}:%{VERSION}-%{RELEASE}\n' "$RPM") # get EVR of the RPM
```

File: pkg-cmp-between.sh [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+RPM_EVR=${RPM_VER/(none)/0} # no epoch will print (none) for the epoch field so we change it to 0
+
+rpmdev-vercmp RPM_EVR EVR1 # check if RPM is newer/equal to EVR1 and store the return val
+RET1=$?
+rpmdev-vercmp RPM_EVR EVR2 # check if RPM is older/equal to EVR2 and store the return val
```
Comment (matyasselmeci) [inline pkg-cmp-between.sh:23]:
```suggestion
rpmdev-vercmp "$RPM_EVR" "$EVR2" # check if RPM is older/equal to EVR2 and store the return val
```

File: pkg-cmp-gt.sh [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+RPM=$1
+EVR=$2
+
+if [[ $# -ne 2 ]]; then
+    echo "Expected 2 args, got $#"
+    echo "$USAGE"
+    exit
+fi
+
+RPM_EVR=$(rpm -q --queryformat '%{EPOCH}:%{VERSION}-%{RELEASE}\n' $RPM) # get EVR of the RPM
```
Comment (matyasselmeci) [inline pkg-cmp-gt.sh:17]:
```suggestion
RPM_EVR=$(rpm -q --queryformat '%{EPOCH}:%{VERSION}-%{RELEASE}\n' "$RPM") # get EVR of the RPM
```

File: pkg-cmp-gt.sh [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+RPM_EVR=${RPM_VER/(none)/0} # no epoch will print (none) for the epoch field so we change it to 0
+
+rpmdev-vercmp RPM_EVR EVR # check if RPM is newer than the EVR
```
Comment (matyasselmeci) [inline pkg-cmp-gt.sh:20]:
```suggestion
rpmdev-vercmp "$RPM_EVR" "$EVR" # check if RPM is newer than the EVR
```

File: pkg-cmp-lt.sh [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+
+USAGE="Usage: pkg-cmp-lt.sh <RPM> <EVR>"
+
+RPM=$1
+EVR=$2
+
+if [[ $# -ne 2 ]]; then
+    echo "Expected 2 args, got $#"
+    echo "$USAGE"
+    exit
```
Comment (matyasselmeci) [inline pkg-cmp-lt.sh:14]:
```suggestion
    exit 2
```

File: pkg-cmp-lt.sh [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+fi
+
+RPM_EVR=$(rpm -q --queryformat '%{EPOCH}:%{VERSION}-%{RELEASE}\n' $RPM) # get EVR of the RPM
```
Comment (matyasselmeci) [inline pkg-cmp-lt.sh:17]:
```suggestion
RPM_EVR=$(rpm -q --queryformat '%{EPOCH}:%{VERSION}-%{RELEASE}\n' "$RPM") # get EVR of the RPM
```

File: pkg-cmp-lt.sh [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+RPM_EVR=${RPM_VER/(none)/0} # no epoch will print (none) for the epoch field so we change it to 0
```
Comment (matyasselmeci) [inline pkg-cmp-lt.sh:18]:
```suggestion
RPM_EVR=${RPM_EVR/(none)/0} # no epoch will print (none) for the epoch field so we change it to 0
```

File: pkg-cmp-lt.sh [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+
+rpmdev-vercmp RPM_EVR EVR # check if RPM is older than the EVR
```
Comment (matyasselmeci) [inline pkg-cmp-lt.sh:20]:
```suggestion
rpmdev-vercmp "$RPM_EVR" "$EVR" # check if RPM is older than the EVR
```

File: pkg-cmp-between.sh [hunk 14]
```diff
@@ -0,0 +1,29 @@
...
+
+USAGE="Usage: pkg-cmp-between.sh <RPM> <EVR1> <EVR2>"
+
+RPM=$1
+EVR1=$2
+EVR2=$3
+
+if [[ $# -ne 3 ]]; then
+    echo "Expected 3 args, got $#"
+    echo "$USAGE"
```
Comment (matyasselmeci) [inline pkg-cmp-between.sh:14]:
```suggestion
    echo "Expected 3 args, got $#" >&2
    echo "$USAGE" >&2
```
Error and usage messages should go to stderr.

File: pkg-cmp-gt.sh [hunk 15]
```diff
@@ -0,0 +1,25 @@
...
+# Exits 0 if RPM is newer than the EVR, 1 otherwise
+
+USAGE="Usage: pkg-cmp-gt.sh <RPM> <EVR>"
+
+RPM=$1
+EVR=$2
+
+if [[ $# -ne 2 ]]; then
+    echo "Expected 2 args, got $#"
+    echo "$USAGE"
```
Comment (matyasselmeci) [inline pkg-cmp-gt.sh:13]:
```suggestion
    echo "Expected 2 args, got $#" >&2
    echo "$USAGE" >&2
```

File: pkg-cmp-lt.sh [hunk 16]
```diff
@@ -0,0 +1,25 @@
...
+# Exits 0 if RPM is older than the EVR, 1 otherwise
+
+USAGE="Usage: pkg-cmp-lt.sh <RPM> <EVR>"
+
+RPM=$1
+EVR=$2
+
+if [[ $# -ne 2 ]]; then
+    echo "Expected 2 args, got $#"
+    echo "$USAGE"
```
Comment (matyasselmeci) [inline pkg-cmp-lt.sh:13]:
```suggestion
    echo "Expected 2 args, got $#" >&2
    echo "$USAGE" >&2
```

**Review by matyasselmeci — DISMISSED** [review]
See suggestions.

File: Dockerfile [hunk 17]
```diff
@@ -42,6 +42,9 @@ RUN \
     # Support old init script dir name
     ln -s /etc/osg/image-{init,config}.d
 
+COPY bin/pkg-cmp-between.sh               /usr/local/bin
+COPY bin/pkg-cmp-gt.sh                    /usr/local/bin
+COPY bin/pkg-cmp-lt.sh                    /usr/local/bin
```
Comment (brianhlin) [inline Dockerfile:47]:
```suggestion
COPY bin/*               /usr/local/bin
```

Docker knows how to use simple wildcards, which allows us to reduce the number of container image layers and thus the size of the final container

**Review by brianhlin — APPROVED** [review]
One minor suggestion but otherwise LGTM. Pre-approving!

**Pull request merged**